Index: impl/src/main/java/uk/ac/manchester/cs/owl/owlapi/OWLLiteralImpl.java
===================================================================
--- impl/src/main/java/uk/ac/manchester/cs/owl/owlapi/OWLLiteralImpl.java	(revision 1891)
+++ impl/src/main/java/uk/ac/manchester/cs/owl/owlapi/OWLLiteralImpl.java	(working copy)
@@ -41,6 +41,7 @@
 import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.io.InputStreamReader;
 import java.util.Arrays;
 import java.util.zip.GZIPInputStream;
 import java.util.zip.GZIPOutputStream;
@@ -119,11 +120,12 @@
 		String decompress(byte[] result) throws IOException {
 			ByteArrayInputStream in = new ByteArrayInputStream(result);
 			GZIPInputStream zipin = new GZIPInputStream(in);
+			InputStreamReader reader = new InputStreamReader(zipin, "UTF-8");
 			StringBuilder b = new StringBuilder();
-			int c = zipin.read();
+			int c = reader.read();
 			while (c > -1) {
 				b.append((char) c);
-				c = zipin.read();
+				c = reader.read();
 			}
 			return b.toString();
 		}
