Index: rdfxml/src/main/java/org/coode/owlapi/rdfxml/parser/TypeRDFSClassHandler.java
===================================================================
--- rdfxml/src/main/java/org/coode/owlapi/rdfxml/parser/TypeRDFSClassHandler.java	(revision 1807)
+++ rdfxml/src/main/java/org/coode/owlapi/rdfxml/parser/TypeRDFSClassHandler.java	(working copy)
@@ -39,7 +39,11 @@
 
 package org.coode.owlapi.rdfxml.parser;
 
+import java.util.Set;
+
 import org.semanticweb.owlapi.model.IRI;
+import org.semanticweb.owlapi.model.OWLAnnotation;
+import org.semanticweb.owlapi.model.OWLClass;
 import org.semanticweb.owlapi.model.UnloadableImportException;
 import org.semanticweb.owlapi.vocab.OWLRDFVocabulary;
 
@@ -59,7 +63,13 @@
 
     public void handleTriple(IRI subject, IRI predicate, IRI object) throws UnloadableImportException {
         // TODO: Change to rdfs:Class? (See table 5 in the spec)
+        if (!getConsumer().getConfiguration().isStrict() && !isAnonymous(subject)) {
+            Set<OWLAnnotation> annos = getConsumer().getPendingAnnotations();
+            OWLClass owlClass = getDataFactory().getOWLClass(subject);
+            addAxiom(getDataFactory().getOWLDeclarationAxiom(owlClass, annos));
+        }
         getConsumer().addClassExpression(subject, false);
         consumeTriple(subject, predicate, object);
     }
+    
 }
