Index: debugging/src/main/java/com/clarkparsia/owlapi/explanation/SingleExplanationGeneratorImpl.java
===================================================================
--- debugging/src/main/java/com/clarkparsia/owlapi/explanation/SingleExplanationGeneratorImpl.java	(revision 1737)
+++ debugging/src/main/java/com/clarkparsia/owlapi/explanation/SingleExplanationGeneratorImpl.java	(working copy)
@@ -42,7 +42,6 @@
         this.reasonerFactory = reasonerFactory;
         this.reasoner = reasoner;
         this.owlOntologyManager = ontology.getOWLOntologyManager();
-        definitionTracker = new DefinitionTracker(ontology);
     }
 
 
@@ -55,6 +54,9 @@
     }
 
     public DefinitionTracker getDefinitionTracker() {
+    	if (definitionTracker == null) {
+            definitionTracker = new DefinitionTracker(ontology);
+    	}
         return definitionTracker;
     }
 
Index: impl/src/main/java/uk/ac/manchester/cs/owl/owlapi/OWLOntologyImpl.java
===================================================================
--- impl/src/main/java/uk/ac/manchester/cs/owl/owlapi/OWLOntologyImpl.java	(revision 1737)
+++ impl/src/main/java/uk/ac/manchester/cs/owl/owlapi/OWLOntologyImpl.java	(working copy)
@@ -1426,8 +1426,12 @@
         OWLOntology other = (OWLOntology) obj;
         return ontologyID.equals(other.getOntologyID());
     }
+    
+    @Override
+    public int hashCode() {
+    	return ontologyID.hashCode();
+    }
 
-
     private class OWLEntityReferenceChecker implements OWLEntityVisitor {
 
         private boolean ref;
