Index: debugging/src/main/java/com/clarkparsia/owlapi/explanation/SingleExplanationGeneratorImpl.java
===================================================================
--- debugging/src/main/java/com/clarkparsia/owlapi/explanation/SingleExplanationGeneratorImpl.java	(revision 1713)
+++ debugging/src/main/java/com/clarkparsia/owlapi/explanation/SingleExplanationGeneratorImpl.java	(working copy)
@@ -42,7 +42,6 @@
         this.reasonerFactory = reasonerFactory;
         this.reasoner = reasoner;
         this.owlOntologyManager = ontology.getOWLOntologyManager();
-        definitionTracker = new DefinitionTracker(ontology);
     }
 
 
@@ -55,6 +54,9 @@
     }
 
     public DefinitionTracker getDefinitionTracker() {
+    	if (definitionTracker == null) {
+            definitionTracker = new DefinitionTracker(ontology);
+    	}
         return definitionTracker;
     }
 
Index: impl/src/main/java/uk/ac/manchester/cs/owl/owlapi/InitVisitorFactory.java
===================================================================
--- impl/src/main/java/uk/ac/manchester/cs/owl/owlapi/InitVisitorFactory.java	(revision 1713)
+++ impl/src/main/java/uk/ac/manchester/cs/owl/owlapi/InitVisitorFactory.java	(working copy)
@@ -253,10 +253,8 @@
 
         @Override
         public Collection<K> visit(OWLDisjointUnionAxiom axiom) {
-            List<OWLClassExpression> list = new ArrayList<OWLClassExpression>(axiom.getClassExpressions());
-            if (named) {
-                deleteAnonymousClasses(list);
-            }
+            List<OWLClassExpression> list = new ArrayList<OWLClassExpression>();
+            list.add(axiom.getOWLClass());
             return (Collection<K>) list;
         }
 
