Index: debugging/src/main/java/com/clarkparsia/owlapi/explanation/BlackBoxExplanation.java
===================================================================
--- debugging/src/main/java/com/clarkparsia/owlapi/explanation/BlackBoxExplanation.java	(.../tags/svn-1602-release-3.1.0)	(revision 21026)
+++ debugging/src/main/java/com/clarkparsia/owlapi/explanation/BlackBoxExplanation.java	(.../trunk)	(revision 21026)
@@ -3,6 +3,7 @@
 import org.semanticweb.owlapi.model.*;
 import org.semanticweb.owlapi.util.OWLAxiomVisitorAdapter;
 import org.semanticweb.owlapi.util.OWLEntityCollector;
+import org.semanticweb.owlapi.reasoner.FreshEntityPolicy;
 import org.semanticweb.owlapi.reasoner.OWLReasoner;
 import org.semanticweb.owlapi.reasoner.OWLReasonerFactory;
 import org.semanticweb.owlapi.apibinding.OWLManager;
@@ -124,7 +125,8 @@
 
     public Set<OWLAxiom> getExplanation(OWLClassExpression unsatClass) {
 
-        if (!getDefinitionTracker().isDefined(unsatClass))
+        if (getReasoner().getFreshEntityPolicy() == FreshEntityPolicy.DISALLOW
+        		&& !getDefinitionTracker().isDefined(unsatClass))
             return Collections.emptySet();
 
         try {
Index: debugging/src/main/java/com/clarkparsia/owlapi/explanation/SingleExplanationGeneratorImpl.java
===================================================================
--- debugging/src/main/java/com/clarkparsia/owlapi/explanation/SingleExplanationGeneratorImpl.java	(.../tags/svn-1602-release-3.1.0)	(revision 21026)
+++ debugging/src/main/java/com/clarkparsia/owlapi/explanation/SingleExplanationGeneratorImpl.java	(.../trunk)	(revision 21026)
@@ -44,7 +44,6 @@
         this.reasonerFactory = reasonerFactory;
         this.reasoner = reasoner;
         this.owlOntologyManager = ontology.getOWLOntologyManager();
-        definitionTracker = new DefinitionTracker(ontology);
     }
 
 
@@ -57,6 +56,9 @@
     }
 
     public DefinitionTracker getDefinitionTracker() {
+    	if (definitionTracker == null) {
+            definitionTracker = new DefinitionTracker(ontology);
+    	}
         return definitionTracker;
     }
 
