Index: debugging/src/main/java/com/clarkparsia/owlapi/explanation/BlackBoxExplanation.java
===================================================================
--- debugging/src/main/java/com/clarkparsia/owlapi/explanation/BlackBoxExplanation.java	(.../tags/svn-1602-release-3.1.0)	(revision 21026)
+++ debugging/src/main/java/com/clarkparsia/owlapi/explanation/BlackBoxExplanation.java	(.../trunk)	(revision 21026)
@@ -3,6 +3,7 @@
 import org.semanticweb.owlapi.model.*;
 import org.semanticweb.owlapi.util.OWLAxiomVisitorAdapter;
 import org.semanticweb.owlapi.util.OWLEntityCollector;
+import org.semanticweb.owlapi.reasoner.FreshEntityPolicy;
 import org.semanticweb.owlapi.reasoner.OWLReasoner;
 import org.semanticweb.owlapi.reasoner.OWLReasonerFactory;
 import org.semanticweb.owlapi.apibinding.OWLManager;
@@ -124,7 +125,8 @@
 
     public Set<OWLAxiom> getExplanation(OWLClassExpression unsatClass) {
 
-        if (!getDefinitionTracker().isDefined(unsatClass))
+        if (getReasoner().getFreshEntityPolicy() == FreshEntityPolicy.DISALLOW
+        		&& !getDefinitionTracker().isDefined(unsatClass))
             return Collections.emptySet();
 
         try {
Index: debugging/src/main/java/com/clarkparsia/owlapi/explanation/SingleExplanationGeneratorImpl.java
===================================================================
--- debugging/src/main/java/com/clarkparsia/owlapi/explanation/SingleExplanationGeneratorImpl.java	(.../tags/svn-1602-release-3.1.0)	(revision 21026)
+++ debugging/src/main/java/com/clarkparsia/owlapi/explanation/SingleExplanationGeneratorImpl.java	(.../trunk)	(revision 21026)
@@ -44,7 +44,6 @@
         this.reasonerFactory = reasonerFactory;
         this.reasoner = reasoner;
         this.owlOntologyManager = ontology.getOWLOntologyManager();
-        definitionTracker = new DefinitionTracker(ontology);
     }
 
 
@@ -57,6 +56,9 @@
     }
 
     public DefinitionTracker getDefinitionTracker() {
+    	if (definitionTracker == null) {
+            definitionTracker = new DefinitionTracker(ontology);
+    	}
         return definitionTracker;
     }
 
Index: MergeStatus.txt
===================================================================
--- MergeStatus.txt	(.../tags/svn-1602-release-3.1.0)	(revision 0)
+++ MergeStatus.txt	(.../trunk)	(revision 21026)
@@ -0,0 +1,12 @@
+
+root=http://smi-protege.stanford.edu/repos/protege/protege4/external/owlapi
+tag=tags/svn-1602-release-3.1.0
+
+Trunk is now based on 
+
+    ${root}/${tag}
+
+Useful commands:
+
+   svn diff ${root}/${tag} ${root}/trunk > owlapi.patch
+       creates a patch file for the official owlapi
Index: mansyntaxparser/src/main/java/org/coode/owlapi/manchesterowlsyntax/ManchesterOWLSyntaxEditorParser.java
===================================================================
--- mansyntaxparser/src/main/java/org/coode/owlapi/manchesterowlsyntax/ManchesterOWLSyntaxEditorParser.java	(.../tags/svn-1602-release-3.1.0)	(revision 21026)
+++ mansyntaxparser/src/main/java/org/coode/owlapi/manchesterowlsyntax/ManchesterOWLSyntaxEditorParser.java	(.../trunk)	(revision 21026)
@@ -922,9 +922,6 @@
                         throw createException(OWLFacet.getFacets().toArray(new String[OWLFacet.getFacetIRIs().size()]));
                     }
                     OWLLiteral con = parseConstant();
-                    if (!con.isRDFPlainLiteral()) {
-                        con = dataFactory.getOWLLiteral(con.getLiteral());
-                    }
                     facetRestrictions.add(dataFactory.getOWLFacetRestriction(fv, con));
                     sep = consumeToken();
                 }
Index: mansyntaxparser/src/main/java/org/coode/owlapi/manchesterowlsyntax/ManchesterOWLSyntaxOntologyParser.java
===================================================================
--- mansyntaxparser/src/main/java/org/coode/owlapi/manchesterowlsyntax/ManchesterOWLSyntaxOntologyParser.java	(.../tags/svn-1602-release-3.1.0)	(revision 21026)
+++ mansyntaxparser/src/main/java/org/coode/owlapi/manchesterowlsyntax/ManchesterOWLSyntaxOntologyParser.java	(.../trunk)	(revision 21026)
@@ -52,10 +52,10 @@
                     br = new BufferedReader(documentSource.getReader());
                 }
                 else if (documentSource.isInputStreamAvailable()) {
-                    br = new BufferedReader(new InputStreamReader(documentSource.getInputStream()));
+                    br = new BufferedReader(new InputStreamReader(documentSource.getInputStream(), "UTF-8"));
                 }
                 else {
-                    br = new BufferedReader(new InputStreamReader(getInputStream(documentSource.getDocumentIRI())));
+                    br = new BufferedReader(new InputStreamReader(getInputStream(documentSource.getDocumentIRI()), "UTF-8"));
                 }
                 StringBuilder sb = new StringBuilder();
                 String line;
Index: api/src/main/java/org/semanticweb/owlapi/io/FileDocumentSource.java
===================================================================
--- api/src/main/java/org/semanticweb/owlapi/io/FileDocumentSource.java	(.../tags/svn-1602-release-3.1.0)	(revision 21026)
+++ api/src/main/java/org/semanticweb/owlapi/io/FileDocumentSource.java	(.../trunk)	(revision 21026)
@@ -1,8 +1,15 @@
 package org.semanticweb.owlapi.io;
 
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.UnsupportedEncodingException;
+
 import org.semanticweb.owlapi.model.IRI;
-
-import java.io.*;
 /*
  * Copyright (C) 2007, University of Manchester
  *
@@ -78,6 +85,11 @@
 
 
     public Reader getReader() {
-        return new InputStreamReader(getInputStream());
+        try {
+            return new InputStreamReader(getInputStream(), "UTF-8");
+        }
+        catch (UnsupportedEncodingException ee) {
+            throw new RuntimeException("Programmer error. Contact the developers at owlapi.sourceforge.net", ee);
+        }
     }
 }
Index: api/src/main/java/org/semanticweb/owlapi/vocab/XSDVocabulary.java
===================================================================
--- api/src/main/java/org/semanticweb/owlapi/vocab/XSDVocabulary.java	(.../tags/svn-1602-release-3.1.0)	(revision 21026)
+++ api/src/main/java/org/semanticweb/owlapi/vocab/XSDVocabulary.java	(.../trunk)	(revision 21026)
@@ -66,7 +66,7 @@
     LANGUAGE("language"),
     NAME("Name"),
     NCNAME("NCName"),
-    NMTOKEN("NMToken"),
+    NMTOKEN("NMTOKEN"),
     ID("ID"),
     IDREF("IDREF"),
     IDREFS("IDREFS"),
Index: api/src/main/java/org/semanticweb/owlapi/util/AbstractOWLOntologyStorer.java
===================================================================
--- api/src/main/java/org/semanticweb/owlapi/util/AbstractOWLOntologyStorer.java	(.../tags/svn-1602-release-3.1.0)	(revision 21026)
+++ api/src/main/java/org/semanticweb/owlapi/util/AbstractOWLOntologyStorer.java	(.../trunk)	(revision 21026)
@@ -68,8 +68,8 @@
                 }
 
 
-
-                BufferedReader br = new BufferedReader(new FileReader(tempFile));
+                BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream(tempFile), "UTF-8"));
+                // BufferedReader br = new BufferedReader(new FileReader(tempFile));
                 BufferedWriter w = new BufferedWriter(new OutputStreamWriter(os, "UTF-8"));
                 String line;
                 while((line = br.readLine()) != null) {

Property changes on: .
___________________________________________________________________
Added: svn:externals
   + MergeStatus.txt http://smi-protege.stanford.edu/repos/protege/protege4/external/owlapi/MergeStatus.txt


